File: javascript/section01/solution05.js
/*
최솟값 구하기 (min)

7개의 수가 주어지면 그 숫자 중 가장 작은 수를 출력하는 프로그램

input : 5 3 7 11 2 15 17
output : 2
*/
let nums = [5, 3, 7, 11, 2, 15, 17]
console.log(`result : ${solution(nums)}`);

function solution(arr) {
    let answer, min = Number.MAX_SAFE_INTEGER;
    
    // 최솟값 구하는 내장함수 이용 : 파라미터로 정수만 가능하고 배열은 전개연산자(...arr)사용해서 넘기기
    // let answer = Math.min(...arr);

    for(let i = 0;  i < arr.length; i++) {
        if(arr[i] < min) {
            min = arr[i];
        }
    }
    answer = min;

    return answer;
}

File: java/java_study/src/section02/Solution05.java
package section02;

import javax.naming.CannotProceedException;

public class Solution05 {

    public static void main(String[] args) {
        /*
        * 소수 (에라토스테네스 체)
        *
        * 자연수 N이 입력되면 1부터 N까지의 소수의 개수를 출력하는 프로그램 작성
        * 만약 20이 입력되면 1부터 20까지의 소수는 2, 3, 5, 7, 11, 13, 17, 19로 총 8개
        * 제한시간 1초
        *
        * input : 20
        * output : 8
        * */

        System.out.println("result : " + solution(20));
    }

    public static int solution(int num) {
        int answer = 0;
        int[] ch = new int[num + 1];

        for (int i = 2; i <= num; i++) {
            if (ch[i] == 0) {
                answer++;
                for (int j = i; j <= num; j += i) {
                    ch[j] = 1;
                }
            }
        }

        return answer;
    }
}
